pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -191,7 +191,8 @@ def processing_fn(results):
 
         if isinstance(obj, qml.QNode):
             if qml.capture.enabled():
-                return self._capture_callable_transform(obj, targs, tkwargs)
+                new_qnode = self.default_qnode_transform(obj, targs, tkwargs)
+                return self._capture_callable_transform(new_qnode, targs, tkwargs)
```

Comment:
I may not understand this correctly but shouldn't we return a qnode here?

Reply:
With capture enabled, a callable is always returned. This was the case before this PR as well. If we return a qnode, then the first primitive in the captured program of a qnode will be the qnode primitive. But, when a transform is applied to a qnode, the first primitive should be the transform's primitive.